### PR TITLE
fix todo: use func GetPreferredNodeAddress get preferred address of node

### PR DESCRIPTION
**Pull request by hanxiaoshuai**
**What this PR does / why we need it**:
fix todo: use func GetPreferredNodeAddress get preferred address of node
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by hanxiaoshuai** [comment]
cc @roberthbailey

**Comment by roberthbailey** [comment]
Thanks for sending a PR to fix this old TODO. I doubt it will break anything, but probably better to merge once 1.10 is cut just to be safe.

/lgtm

File: test/e2e/framework/util.go [hunk 1]
```diff
@@ -3200,16 +3201,11 @@ func UpdateDaemonSetWithRetries(c clientset.Interface, namespace, name string, a
...
 	hosts := []string{}
+	addrTypes := []v1.NodeAddressType{addrType}
 	for _, n := range nodelist.Items {
-		for _, addr := range n.Status.Addresses {
-			// Use the first external IP address we find on the node, and
-			// use at most one per node.
-			// TODO(roberthbailey): Use the "preferred" address for the node, once
-			// such a thing is defined (#2462).
-			if addr.Type == addrType {
-				hosts = append(hosts, addr.Address)
```
Comment (deads2k) [inline test/e2e/framework/util.go]: This lists all that match, the new function lists one per addrtype.  Who determined this was safe?

**Pull request closed without merging**